### PR TITLE
Migrate restore tests to new structure (part 1)

### PR DESCRIPTION
**Pull request by skriss**
Here's the first batch of migrated `pkg/restore` tests. Probably easiest to review one commit at a time.

These commits don't need to be squashed for merge.

cc @nrb @prydonius @carlisia

### CODE REVIEW

**Comment by skriss** [comment]
@nrb PTAL, thanks!

File: pkg/restore/restore_new_test.go [hunk 1]
```diff
@@ -109,8 +569,148 @@ func TestRestoreNew(t *testing.T) {
...
+		)
+
+		assertEmptyResults(t, warnings, errs)
+		assertResourceCreationOrder(t, tc.resourcePriorities, recorder.resources)
+	}
+}
+
+// assertResourceCreationOrder ensures that resources were created in the expected
+// order. Any resources *not* in resourcePriorities are required to come *after* all
+// resources in , in any order.
```
Comment (nrb) [inline pkg/restore/restore_new_test.go]: Super minor, but `in , in` is a typo.

File: pkg/restore/restore_new_test.go [hunk 2]
```diff
@@ -109,8 +569,148 @@ func TestRestoreNew(t *testing.T) {
...
+	// we saw created. Once we've seen a resource in 'resourcePriorities', we should
+	// never see another instance of a prior resource.
+	lastSeen := 0
+
+	// Find the index in 'resourcePriorities' of the resource type for
+	// the current item, if it exists. This index ('current') *must*
+	// be greater than or equal to 'lastSeen', which was the last resource
+	// we saw, since otherwise the current resource would be out of order. By
+	// initializing current to len(ordered), we're saying that if the resource
+	// is not explicitly in orederedResources, then it must come *after*
```
Comment (nrb) [inline pkg/restore/restore_new_test.go]: `orederedResources` -> `orderedResources`

**Comment by skriss** [comment]
Thanks, fixed. The commits don't need to be squashed for merge.

**Pull request merged**